strings/camel_case_to_snake_case.py

```diff
@@ -0,0 +1,83 @@
+"""
+this function focuses on converting camel case (eg: camelCase)
+to snake case (snake_case)
+"""
+
+
+def to_snake_case(word: str) -> str:
+    """
+    This to_snake_case function convert camel case words (eg: camelCase)
+    to snake case (eg: snake_case)
+    >>> to_snake_case('helloWorld')
+    'hello_world'
+    >>> to_snake_case('camelCase')
+    'camel_case'
+    >>> to_snake_case('hacktoberFest')
+    'hacktober_fest'
+    >>> to_snake_case('')
+    Traceback (most recent call last):
+    Exception: Provide camel case eg: helloWorld
+    >>> to_snake_case('HelloWorld')
+    Traceback (most recent call last):
+    Exception: Provide camel case eg: helloWorld
+    >>> to_snake_case(' ')
+    Traceback (most recent call last):
+    Exception: Provide camel case eg: helloWorld
+    """
+
+    """
+    Raising the error if inputted data is not in the required format
+    """
+
+    # checks if the word has capital character in it,
+    # and first character is not the capital.
+    if word != word.lower() and word[0] == word[0].lower():
+        snake_case = ""
+        index_list = []
+        # check number of Capital letter and append their indexes in list
+        for character in word:
+            if character != character.lower():
+                index_list.append(word.index(character))
+
+        # append one more element
+        index_list.append(100)
+
+        # initializing few variables
+        i = 0
+        temp = 1
+
+        # changing original string to lowercase
+        word = word.lower()
+
+        for index in index_list:  # iterates over list of indexes
```

Comment:
You're iterating through the index list and getting the element itself. 

You then don't use this element, but instead use `i`, the previous element.

Instead of doing this, you can loop over the length of the list and index to the item of n-1
eg 
```py
for n in range(len(index_list)):
    index_list[n-1]
```
instead of
```py
i = 0
temp = 1
for item in index_list:
    index_list[i]
    i = item
    if temp > ...:
       ...
    elif temp < ...:
       ...
    temp += 1
```
Note this also gets rid of the variable `temp` and both if statements, condensing code and making it more understandable

Reply:
Here, I use variables "i" and "index" for string slicing and "index" is reassigned to "i" as if there are multiple uppercase characters, temp variable is used to avoid adding an underscore at the end of the modified string.